src/server/mutate.rs

```diff
@@ -53,33 +54,39 @@ async fn mutate_internal(review: AdmissionReview) -> Result<impl warp::Reply, wa
         return Ok(warp::reply::with_status(
             warp::reply::json(&error_response),
             warp::http::StatusCode::BAD_REQUEST,
-        ));
-    };
-
-    let patch = r#"[{
-        "op": "add",
-        "path": "/spec/runtimeClassName",
-        "value": "edera"
-    }]"#;
-    let patch = BASE64_STANDARD.encode(patch);
-
-    let response = AdmissionReviewResponse {
-        api_version: "admission.k8s.io/v1".to_string(),
-        kind: "AdmissionReview".to_string(),
-        response: Some(Response {
-            uid: request.uid,
-            allowed: true,
-            patch_type: Some("JSONPatch".to_string()),
-            patch: Some(patch),
-        }),
+        ))
     };
+    // Only do an update operation if there is
+    // no class set already 
+    if request.runtimeclass.is_empty() {
+        let patch = r#"[{
+            "op": "add",
+            "path": "/spec/runtimeClassName",
+            "value": "edera"
+        }]"#;
+        let patch = BASE64_STANDARD.encode(patch);
+
+        let response = AdmissionReviewResponse {
+            api_version: "admission.k8s.io/v1".to_string(),
+            kind: "AdmissionReview".to_string(),
+            response: Some(Response {
+                uid: request.uid,
+                allowed: true,
+                patch_type: Some("JSONPatch".to_string()),
+                patch: Some(patch),
+            }),
+        };
 
-    info!("mutating {}/{}", request.namespace, request.name);
-    debug!("payload {:?}", response);
-    Ok(warp::reply::with_status(
-        warp::reply::json(&response),
-        warp::http::StatusCode::OK,
-    ))
+        info!("mutating {}/{}", request.namespace, request.name);
+        debug!("payload {:?}", response);
+        return Ok(warp::reply::with_status(
+            warp::reply::json(&response),
+            warp::http::StatusCode::OK,
+        ))
+    } else {
+        // schristoff(TODO): is it ok?
+        return Ok(())
```

Comment:
I think this needs to return a warp reply (like above) as OK with a json patch that's empty (eg. `[]`).. I don't see much information on a quick scan [here](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#response) but that's my guess is kubernetes will treat it as a no-op.